osu.Game/Screens/OsuScreen.cs

```diff
@@ -81,7 +81,12 @@ public abstract class OsuScreen : Screen, IOsuScreen, IHasDescription
 
         public virtual float BackgroundParallaxAmount => 1;
 
-        public virtual bool AllowTrackAdjustments => true;
+        [Resolved]
+        private MusicController musicController { get; set; }
+
+        private bool? allowTrackAdjustments;
+
+        public virtual bool AllowTrackAdjustments => allowTrackAdjustments ??= (musicController?.AllowTrackAdjustments ?? false);
```

Comment:
how does this work? doesn't this change break the entire mechanism? it's certainly not what i had in mind when suggesting moving the fallback logic to the screen.

in my mind the virtual would _still_ be `bool?`, but there would be a *separate* member combining it with the music controller's value for reading.

Reply:
if not overriden in the derived screen, this should remember the `MusicController`'s value first time it's queried (first time it's pushed to the stack) and give back this value on later calls (when getting back on screen if the screen above exited). Though right now my test is falilng because apparently `musicController` just doesn't get resolved in the test screen, trying to figure out what's going on

Reply:
you probably need to cache one locally in the test. compare `TestSceneNowPlayingOverlay` for example.

Reply:
tried to do something with the test, seems to work i guess

Reply:
I've made some (non-minor) refactoring to make this a bit cleaner. The way you wrote it is quite an interesting solution, but as per the conversation above, it's *very* hard to mentally parse.

I've restored the nullable portion you had, but moved the setting/reading `MusicController` logic to be within `OsuScreen`. This avoids the read being done in a different place to the set.

In addition, the stored value is now done so at suspend time, rather than enter, which is probably a more expected behaviour (in the case that a screen changed the value on `MusicController` directly during its execution). I don't believe we rely on this in any way at the moment. @smoogipoo also mentioned that this change rightfully will handle nested `ScreenStack`s better than before.